GitHub action 세팅입니다.

KTH-96 opened the pull request:
- java, gradle 세팅

KTH-96 committed:
feat: github action start

- java, gradle set up

KTH-96 committed:
fix: github action

- jdk set up uses 부분 추가
- push main branch 추가, pull request 전 브랜치에 적용

KTH-96 committed:
fix: 오타수정

- action -> actions

KTH-96 committed:
fix: gradlew is not executable에러 수정

- 실행권한 부분 추가

f-lab-lyan commented on .github/workflows/github-action.yml:
```diff
@@ -0,0 +1,28 @@
+          java-version: '17'
+          distribution: 'corretto'
+
+      - name: chmod to make gradlew executable
+        run: chmod +x ./gradlew
```
이 줄은 꼭 필요한 것인가요?

KTH-96 replied:
밑에 build를 하는 부분에서 gradlew의 실행권한이 없어 빌드가 되지않는 문제가 있어 추가했습니다.

f-lab-lyan replied:
https://stackoverflow.com/questions/6476513/git-file-permissions-on-windows

f-lab-lyan commented on .github/workflows/github-action.yml:
```diff
@@ -0,0 +1,28 @@
+      - name: set up gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
```
테스트 용이니까, `build`보다는 `test`가 더 좋지 않을까 싶은데요?

KTH-96 replied:
test를 사용해도 build.gradle로 프로젝트를 초기화 한다고 하는군요! 수정하겠습니다.
gradle test 명령어를 실행하면 테스트 관련 코드들만 컴파일하고 끝나는 작업이라 생각했습니다.

f-lab-lyan commented on .github/workflows/github-action.yml:
```diff
@@ -0,0 +1,28 @@
+name: github-action
+on:
+  push:
+    branches: [ "main", "github-action" ]
```
`"github-action"`이라는 브랜치는 여기에는 필요없지 않을까요?

KTH-96 replied:
pr을 날리기 이전에 github action이 성공하는지 확인하고자 적어 놓은 부분인데 지워야 할 거 같습니다.

KTH-96 committed:
refactor github-action.yml 수정

- window에서 gradlew 실행권한 주는 코드 삭제
- git-action push branch 제거
- gradle build 에서 test로 변경
